### PR TITLE
Add codeql-version to outputs

### LINKED ISSUE #2367: Add CodeQL version to github/codeql-action/init outputs
Hi there,

I'm working on caching some query compilations and it'd be helpful to receive the CodeQL version as output from the `github/codeql-action/init` action. This could be done similar to the `codeql-path` output. It appears that this would require minimal changes in the following locations:

https://github.com/github/codeql-action/blob/b611370bb5703a7efb587f9d136a52ea24c5c38c/src/init-action.ts#L557

Add something like `core.setOutput("codeql-version", await codeql.getVersion()).version)`.

And:

https://github.com/github/codeql-action/blob/b611370bb5703a7efb587f9d136a52ea24c5c38c/init/action.yml#L139-L141

Add something like:

```yaml
   codeql-version: 
     description: The version of the CodeQL binary used for analysis 
```

### PR DESCRIPTION
**Pull request by mschwager**
Fixes #2367.

### Merge / deployment checklist

- [x] Confirm this change is backwards compatible with existing workflows.
- [X] Confirm the [readme](https://github.com/github/codeql-action/blob/main/README.md) has been updated if necessary.
- [x] Confirm the [changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md) has been updated if necessary.

Regarding the CHANGELOG check above, my understanding was that this would be [automatically populated on release](https://github.com/github/codeql-action/blob/main/CONTRIBUTING.md#releasing-write-access-required)?

### CODE REVIEW

**Comment by aeisenberg** [comment]
Thanks for your PR! The changelog note for this PR needs to be added manually. The automation for the changelog is around adding the release version number and the date.

**Comment by mschwager** [comment]
> Thanks for your PR! The changelog note for this PR needs to be added manually. The automation for the changelog is around adding the release version number and the date.

Done!

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ Note that the only difference between `v2` and `v3` of the CodeQL Action is the
 ## [UNRELEASED]
 
 - Improve the reliability and performance of analyzing code when analyzing a compiled language with the `autobuild` [build mode](https://docs.github.com/en/code-security/code-scanning/creating-an-advanced-setup-for-code-scanning/codeql-code-scanning-for-compiled-languages#codeql-build-modes) on GitHub Enterprise Server. This feature is already available to GitHub.com users. [#2353](https://github.com/github/codeql-action/pull/2353)
+- Add `codeql-version` to outputs [#2368](https://github.com/github/codeql-action/pull/2368)
```
Comment (aeisenberg) [inline CHANGELOG.md:10]:
```suggestion
- Add `codeql-version` to outputs. [#2368](https://github.com/github/codeql-action/pull/2368)
```

**Review by aeisenberg — DISMISSED** [review]

**Comment by mschwager** [comment]
Hey @ginsbach or @aeisenberg, would you mind approving the CI workflows to unblock merging?

**Comment by mschwager** [comment]
Looks like it needs re-approval after the merge too

**Review by aeisenberg — APPROVED** [review]

**Comment by aeisenberg** [comment]
Thanks for the reminder. And thanks for your contribution!

**Comment by mschwager** [comment]
Sure thing, thanks for the speedy reviews!

**Pull request merged**